tests/generative/data_strategies.py

```diff
@@ -45,13 +44,15 @@ def patient_records(draw, class_, schema, value_strategies):
     # This strategy ensures that the patient ids are unique. We need to maintain the state to ensure that uniqueness
     # inside the strategy itself so that we can ensure the tests are idempotent as Hypothesis requires. That means that
     # this strategy must be called once only for a given table in a given test.
-    used_ids = []
+
+    # patients IDs are a permutation of the unique integers representing all possible patient IDs,
+    # between 1 and max_patient_id.  We pop these one at a time to create patient records, so
+    # somewhere between 0 and all of them will be used in the patient_records strategy
+    patient_ids = draw(st.permutations(list(range(1, max_patient_id + 1))))
 
     @st.composite
     def one_patient_record(draw_):
-        id_ = draw_(patient_ids)
-        hyp.assume(id_ not in used_ids)
-        used_ids.append(id_)
+        id_ = patient_ids.pop()
         return draw(record(class_, st.just(id_), schema, value_strategies))
```

Comment:
This is a great idea.

I think you can simplify `one_patient_record()`:

```
def one_patient_record():
        id_ = patient_ids.pop()
        return record(class_, st.just(id_), schema, value_strategies)
```

Reply:
Nope, when I tried this it started generating duplicate ids and getting SQL integrity errors

Reply:
I did need to change `return draw(record(...` to `return draw_(record(...` to make it use the inner function's `draw_` and avoid the warning about using `st.composite` without using draw internally though